__tests__/unit-tests/existingRegionMiddleware.test.js

```diff
@@ -0,0 +1,35 @@
+const existingRegionMiddleware = require('../../src/middlewares/existingRegionMiddleware');
+const path = require('path');
+
+describe('existingRegionMiddleware', () => {
+    const res = {};
+    const next = jest.fn();
+
+    res.status = jest.fn().mockReturnValue(res);
+    res.sendFile =  jest.fn()
+
+    it.each([
+        [{params: {regionId: "asia"}}],
+        [{params: {regionId: "americas"}}],
+        [{params: {regionId: "europe"}}],
+    ])('should pass control to the next middleware when a valid region Id has been sent in the request', (req) => {
+        existingRegionMiddleware(req, res, next);
+        expect(next).toHaveBeenCalled()
```

Comment:
Nice! That has just taught me a whole new spectrum of testing possibilities with this testing library. Thank you!

Reply:
You're welcome 😁.  I'll jump on some more testing shortly.